predictor_pol/flask_app.py

```diff
@@ -98,52 +97,71 @@ def main():
                     )
 
         if validate(session):
-            answer_id = save_response(session)
+            predictions = predict(session)
+            answer_id = save_response(session, predictions['candidate_id'])
             session['answer_id'] = answer_id
-            return render_template('success.html')
+
+            return render_template('success.html',
+                                   predicted_candidate_name=predictions['candidate_name'],
+                                   candidatos = CANDIDATOS)
         else:
             # Esto también
             return 'Error'
 
     session.clear()
     session['page'] = 0
     return render_template(
-        'main.html',
-        candidatos=CANDIDATOS
+        'main.html'
     )
 
 
 def validate(form):
     valid_keys = {
-        'candidato',
         *_get_question_keys(PREGUNTAS)
     }
     return all(form.get(key, '').isdecimal() for key in valid_keys)
 
 
 def predict(responses):
-    xgb = joblib.load('xg_model')
-    df_test = pd.DataFrame.from_dict({'resp1': [1], 'resp2': [1]})
-    print(xgb.predict(df_test))
+    candidate_model = joblib.load(PATH + 'candidate_model.joblib')
+    pca = joblib.load(PATH + 'pca.joblib')
+
+    d = {}
+    for i in range(1,27):
```

Comment:
Would it be better not to hard code it to 27? Maybe calculate the total number of questions in the previous line (or even better in a constant) and then iterate in its range.

This could be calculated in one line with something like `sum([len(category) for category in QUESTIONS])`